docs/developer/feeds.rst

```diff
@@ -4,122 +4,199 @@
 
-``subscribe_to_feed`` by default protects against an agent subscribing to the
-same feed multiple times. This is in case you want to run a client script
-multiple times without restarting the agent, which can call
-``subscribe_to_feed`` multiple times for the same feed. If you purposefully want
-to subscribe to a feed more than once, you can set ``force_subscribe=True``.
+    agent.publish_to_feed(feed_name, message)
 
+For standard feeds, message can be any json-ifyable object (i.e. strings, ints,
+floats, bools, or dicts containing these).
+Callbacks will receive the tuple ``(message, feed_data)`` where ``feed_data``
+is a dict encoding most OCS.Feed attributes.
+
+
+.. _recorded_feed_registration:
+
+Recorded Feed Registration
+````````````````````````````
+To make sure that a feed is picked up
+by the aggregator, it must be registered with the option 'record=True'.
+It also must be registered with the frame_length, which tells the aggregator
+how long each frame should be in seconds.
+An example can be seen in LS372_agent.py::
+
+    agg_params = {
+        'frame_length': 10*60 #[sec]
+    }
+    self.agent.register_feed('temperatures',
+                             record=True,
+                             agg_params=agg_params,
+                             buffer_time=1)
+
+.. _feed_message_format:
+
+Recorded Feed Message Format
+`````````````````````````````
+Recorded feeds require data to have a specific structure so that the aggregator
+can encode the data into G3 objects.
+Each message published should contain a ``block`` of data, or a group of data
+that is co-sampled. Messages should have this structure::
+
+    message = {
+        'block_name': <Key to identify group of co-sampled data>
+        'timestamp': <ctime of data>
+        'data': {
+            'field_name_1': <datapoint1>,
+            'field_name_2': <datapoint2>
+        }
+    }
+
+or if data is buffered on the agent-side, multiple co-sampled data points can
+be passed at once as::
+
+    message = {
+        'block_name': <Key to identify group of co-sampled data>
+        'timestamps': [ctime1, ctime2, ... ]
+        'data': {
+            'field_name_1': [data1_1, data1_2, ...],
+            'field_name_2': [data2_1, data2_2, ...]
+        }
+    }
+
+Note the pluralized ``timestamps`` key.
+
+Data with consistent ``block_names`` will be written to disk as a single
+``G3TimesampleMap`` object, which stores co-sampled data as a map containing
+multiple G3Vector objects along with a vector of timestamps.
+In the example above, the keys of the ``G3TimesampleMap`` will be
+``field_name_1`` and ``field_name_2``.
+The ``block_name`` is only used internally and will not be written
+to disk.
+
+Each set of data that a feed publishes that is non-cosampled should be
+published to a different ``block_name``.
+For instance, for the L372 agent data coming from separate channels are not
+co-sampled.
+The LS372 temperatures should then look like::
```

Comment:
The 372 records temperature and resistance. The 240's have a voltage reading. I'd switch V to R here and keep the 372 example.